evalml/automl/pipeline_search_plots.py

```diff
@@ -1,19 +1,18 @@
 from evalml.utils import import_or_raise, jupyter_check
 
 
-class SearchIterationPlot():
-    def __init__(self, data, show_plot=True):
+class SearchIterationPlot:
```

Comment:
Before we kept a reference to the latest results by passing a reference to AutoMLSearch. In order to not have a self-reference but still be able to access the latest results, I'm adding results as an argument to init and update. The `plot` property of `AutoMLSearch` will always pass the latest results to the plot.